packages/driver/src/cross-origin/patches/fetchAndXMLHttpRequest.ts

```diff
@@ -0,0 +1,102 @@
+const captureFullRequestUrl = (relativeOrAbsoluteUrlString, window) => {
+  // need to pass the window here by reference to generate the correct absolute URL if needed. Spec Bridge does NOT contain sub domain
+  let url
+
+  try {
+    url = new URL(relativeOrAbsoluteUrlString).toString()
+  } catch (err1) {
+    try {
+      // likely a relative path, construct the full url
+      url = new URL(relativeOrAbsoluteUrlString, window.location.origin).toString()
+    } catch (err2) {
+      return undefined
+    }
+  }
+
+  return url
+}
+
+export const patchXmlHttpRequest = (Cypress, window) => {
+  // intercept method calls and add cypress headers to determine cookie applications in the proxy
+  // for simulated top
+
+  if (Cypress.config('experimentalSessionAndOrigin')) {
```

Comment:
nit: can we return early instead?

```
if (!Cypress.config('experimentalSessionAndOrigin')) {
  return
}
```

Reply:
yep I can refactor both the fetch and xmlHttpRequest to just short circuit!

Reply:
addressed in [498a311](https://github.com/cypress-io/cypress/pull/23822/commits/498a311174e06968a390a258dbe5629bfa2acc2d)